### DISCUSSION TITLE
Zero click Cloud Teleport installation on edge devices running k3s (SSH and kubectl)

### DISCUSSION THREAD

**Discussion by Shaked** [gravitational/teleport#27846] [Q&A]
Hi, I have a process that automatically installs my edge devices (hundreds of). Part of the process includes a step that runs an ansible playbook locally, which also installs k3s, flux and few more tools. I want to add Teleport Cloud to this automation so that I will be able to use the teleport `ssh` and `kubectl` access (i.e. it should work on OS level and k8s level). 

I have managed to do that with a manual process, which required me to use `tsh login ... <my_user>`. but for automation I cannot use my own user instead I need some sort of service account and this SA should either have a non expire login creds or automatic creds generation "on the fly". 

What will be the right way to achieve this?

**Comment by dgkirkwood** [comment]
Hi @Shaked,

Can you elaborate on the goal here? 

Would you like your edge devies to all register to a single Teleport Cloud tenant? 
If so, this can be done by creating a join token ahead of time. This token can be created to allow SSH and Kubernetes access and can have a configurable lifetime. 

See more on this process here - https://goteleport.com/docs/management/join-services-to-your-cluster/join-token/

If you have other constraints or a different goal let me know!

Dan.

**Reply by Shaked** [reply]
Hi @dgkirkwood, thanks for this! It's definitely what I'm looking for, one thing I am not sure about - is how do I use an automated service account instead of my own user to login? What I mean is that I'd like to run `tsh login --proxy=teleport.example.com --user=email@example.com` automatically through something like a CICD system

**Reply by dgkirkwood** [reply]
OK yep I see. The best way to achieve this with CICD is to use Machine ID. 

You can find guides for using Teleport in pipelines for Github Actions, Gitlab, Jenkins and more here: https://goteleport.com/docs/machine-id/introduction/#guides
